tests/foreman/ui_airgun/test_ldap_authentication.py

```diff
@@ -241,3 +311,343 @@ def test_positive_create_with_idm_org_and_loc(session, ipa_data):
+            'external_groups.auth_source': 'LDAP-' + auth_source.name,
+        })
+        assert session.usergroup.search(ldap_usergroup_name)[0]['Name'] == ldap_usergroup_name
+        session.organization.select(ANY_CONTEXT['org'])
+        session.location.select(ANY_CONTEXT['location'])
+        session.user.update(ldap_data['ldap_user_name'], {'user.auth': 'LDAP-' + auth_source.name})
+        with Session(
+                test_name,
+                ldap_data['ldap_user_name'],
+                ldap_data['ldap_user_passwd'],
+        ) as ldapsession:
+            ldapsession.location.create({'name': location_name})
+            assert ldapsession.location.search(location_name)[0]['Name'] == location_name
+            current_user = ldapsession.location.read(location_name)['current_user']
+            assert current_user == ldap_data['ldap_user_name']
+        session.usergroup.update(
+            ldap_usergroup_name, {'roles.resources.unassigned': [foreman_role]})
+    with Session(
+            test_name,
+            ldap_data['ldap_user_name'],
+            ldap_data['ldap_user_passwd'],
+    ) as ldapsession:
+        with raises(NavigationTriesExceeded):
+            ldapsession.location.create({'name': gen_string('alpha')})
+
+
+@tier2
+def test_positive_update_external_user_roles(
+        session, ldap_data, ldap_user_name, test_name, auth_source, ldap_usergroup_name):
+    """Assure that user has roles/can access feature areas for
+    additional roles assigned outside any roles assigned by his group
+
+    :id: a487f7d6-22f2-4e42-b34f-8d984f721c83
+
+    :setup: Assign roles to UserGroup and configure external UserGroup
+        subsequently assign specified roles to the user(s).  roles that are
+        not part of the larger UserGroup
+
+    :steps:
+        1. Create an UserGroup.
+        2. Assign some roles to UserGroup.
+        3. Create an External AD UserGroup as per the UserGroup name in AD.
+        4. Assign some more roles to a User(which is part of external AD
+           UserGroup) at the User level.
+        5. Login to sat6 with the above AD user and attempt to access areas
+           assigned specifically to user.
+
+    :expectedresults: User can access not only those feature areas in his
+        UserGroup but those additional feature areas / roles assigned
+        specifically to user
+
+    :CaseLevel: Integration
+    """
+    foreman_role = gen_string('alpha')
+    katello_role = gen_string('alpha')
+    ak_name = gen_string('alpha')
+    location_name = gen_string('alpha')
+    with session:
+        session.role.create({'name': foreman_role})
+        session.filter.create(
+            foreman_role,
+            {
+                'resource_type': 'Location',
+                'permission.assigned': PERMISSIONS['Location'],
+            }
+        )
+        session.usergroup.create({
+            'usergroup.name': ldap_usergroup_name,
+            'roles.resources.assigned': [foreman_role],
```

Comment:
perhaps we should add this permission to user and not to usergroup

Reply:
No, looks like we both misread test steps - usergroup should have some permissions assigned (foreman permission in this case), then user should be updated with extra permissions (katello permisions in this particular test). Now test steps seem ok to me.